walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,8 +14,59 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    long duration = request.getDuration();
+    int[] minutes = new int[1441];
+    List<TimeRange> res = new ArrayList<>();
+    List<String> requestAttendees = new ArrayList(request.getAttendees());
+
+    for (Event event : events) {
+      int rangeStart = event.getWhen().start();
+      int rangeEnd = event.getWhen().end();
+
+      List<String> eventAttendees = new ArrayList(event.getAttendees());
+
+      int mustFill = 0;
+
+      for (String attendee : eventAttendees) {
+        if (requestAttendees.contains(attendee)) {
+          mustFill = 1;
+          break;
+        }
+      }
+      if (mustFill == 1) {
+        for (int i = rangeStart; i < rangeEnd; i++) {
+          minutes[i] = 1;
+        }
+      }
+    }
+
+    for (int i = 0; i < 1440; i++) {
+      if (minutes[i] == 0) {
+        int curStart = i;
+        int curEnd = i;
+
+        for (int j = i; j < 1440; j++) {
```

Comment:
Why do you need to check the whole day? You can limit it to i + duration and 1440.

Reply:
The task asks for the largest range that fits the duration of meeting request. Hence, I need to search largest interval.
But I will update the first for loops iterator. So, the total complexity is still O(MINUTES_IN_ONE_DAY).

Reply:
Ah, sorry I didn't read the task properly